lib/java/com/google/android/material/lists/viewHolders/MaterialViewHolder.java

```diff
@@ -0,0 +1,63 @@
+package com.google.android.material.lists.viewHolders;
+
+import android.content.res.Resources;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+
+import androidx.annotation.DimenRes;
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.material.R;
+
+/** A standard view holder that all specific Material view holders extend */
+public abstract class MaterialViewHolder extends RecyclerView.ViewHolder {
+
+    protected MaterialViewHolder(@NonNull View itemView) {
+        super(itemView);
+    }
+
+    protected MaterialViewHolder(int layout, @NonNull ViewGroup parent, @DimenRes int insetDividerXRes) {
+
+      this(LayoutInflater.from(parent.getContext()).inflate(layout, parent, false));
+      initializeDivider(insetDividerXRes);
+    }
+
+    View divider;
+    private void initializeDivider(@DimenRes int insetDividerXRes){
+
+      insetDividerX = itemView.getContext().getResources().getDimension(insetDividerXRes);
+      divider = itemView.findViewById(R.id.material_list_item_divider);
+
+      ViewGroup.LayoutParams params = divider.getLayoutParams();
+      params.width = fullBleedDividerWidth;
+      divider.setLayoutParams(params);
+      this.insetDividerX = insetDividerX * this.itemView.getContext().getResources().getDisplayMetrics().density;
+
+    }
```

Comment:
This made me curious.
What's the reason for not using `RecyclerView.ItemDecoration` for adding dividers (there's probably a good reason)?

Reply:
> In addition, the nature of a divider is also complex. After considering a certain transparency, width and dark theme support, you are now spending a significant amount of time beautifying what is essentially a thin noodle 🍜
 
Great question! A specialized "Material divider" can handle different needs a certain divider may have. 

Supported Divider widths:
* Full-bleed
* Inset
* None

Supported Divider colors:
* Adapted to use transparency (and work with Material theming)
* Supports dark mode

Additions:
* More performant drawing (calculates the location of the divider instead of using a full-blown RelativeLayout or ConstraintLayout)
* Proper Material spec dimensions

A simple RecyclerView ItemDecoration only does full-bleed and does not address the additions above. Especially the world of dark theming, I still see a lot of popular apps that do not support dark themed dividers. Having such could remove this issue from tons of apps.

Reply:
RecyclerView.ItemDecoration is more like a band-aid solution 🩹

Reply:
That sounds as compelling reasons. Thanks for the comprehensive answer.

But let's discuss "the noodle" a bit.
I'm not a theming expert in any way, but I just thought that `RecyclerView.ItemDecoration` was there for a specific reason.
Also I usually do insets with a drawable (which I always thought would play nicely with my theme - I haven't had the pleasure to implement support for darkmode yet).

Example of insets using a custom drawable:

Drawable with inset:
```xml
<inset xmlns:android="http://schemas.android.com/apk/res/android"
       android:insetLeft="24dp"
       android:insetRight="24dp" >

    <shape>
        <size android:height="1dp"/>
        <solid android:color="@color/recyclerview_divider" />
    </shape>
</inset>
```
The code required to draw insets on your listitems (if I remember `java` correctly):
```java
DividerItemDecoration decorator = new DividerItemDecoration(context, LinearLayoutManager.VERTICAL);
decorator.setDrawable(drawable);
recyclerView.addItemDecoration(decorator);
```

You could easily replace DividerItemDecoration with more specific implementations (like material versions of `RecyclerView.ItemDecoration`). But the above example is actually very flexible by only depending on the drawable used as your divider. 

Another question is also, where does this feature really belong and should the `MaterialViewHolder` really have this responsibility? I tend to think it shouldn't.